sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -82,47 +97,76 @@ public ServiceBusMessage(byte[] body) {
      * @throws NullPointerException if {@code receivedMessage} is {@code null}.
      */
     public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage) {
-        this.body = receivedMessage.getBody();
+        Objects.requireNonNull(receivedMessage, "'receivedMessage' cannot be null.");
+
+        this.amqpAnnotatedMessage = new AmqpAnnotatedMessage(receivedMessage.getAmqpAnnotatedMessage());
         this.context = Context.NONE;
-        setMessageId(receivedMessage.getMessageId());
-        setScheduledEnqueueTime(receivedMessage.getScheduledEnqueueTime());
-        setContentType(receivedMessage.getContentType());
-        setCorrelationId(receivedMessage.getCorrelationId());
-        setLabel(receivedMessage.getLabel());
-        setPartitionKey(receivedMessage.getPartitionKey());
-        setReplyTo(receivedMessage.getReplyTo());
-        setReplyToSessionId(receivedMessage.getReplyToSessionId());
-        setTimeToLive(receivedMessage.getTimeToLive());
-        setTo(receivedMessage.getTo());
-        setSessionId(receivedMessage.getSessionId());
-        setViaPartitionKey(receivedMessage.getViaPartitionKey());
+
+        final AmqpDataBody amqpDataBody = ((AmqpDataBody) receivedMessage.getAmqpAnnotatedMessage().getBody());
+        final List<BinaryData> binaryDataList = amqpDataBody.getData().stream().collect(Collectors.toList());
+        if (binaryDataList.size() > 0) {
+            binaryData = binaryDataList.get(0).getData();
+        } else {
+            binaryData = EMPTY_BYTE_ARRAY;
+        }
+
+        // clean up data which user is not allowed to set.
+        amqpAnnotatedMessage.getHeader().setDeliveryCount(null);
+
+        removeValues(amqpAnnotatedMessage.getMessageAnnotations(), LOCKED_UNTIL_KEY_ANNOTATION_NAME,
```

Comment:
This may be dealt with in a copy constructor for AmqpAnnotatedMessage.

Reply:
Like we spoke in teams, I will leave it here because ` it seems weird a copy constructor wouldn't copy everything.`